test/helpers/deployZNS.ts

```diff
@@ -61,27 +73,40 @@ export const deployPriceOracle = async ({
   registrationFee : BigNumber;
 }) : Promise<ZNSPriceOracle> => {
   const priceOracleFactory = new ZNSPriceOracle__factory(deployer);
-  const priceOracle = await priceOracleFactory.deploy();
 
-  await priceOracle.initialize(
-    accessControllerAddress,
-    priceConfig,
-    registrationFee
-  );
+  const priceOracle = await upgrades.deployProxy(
+    priceOracleFactory,
+    [
+      accessControllerAddress,
+      priceConfig,
+      registrationFee,
+    ],
+    {
+      kind: "uups",
+    }
+  ) as ZNSPriceOracle;
 
   return priceOracle;
 };
 
 export const deployDomainToken = async (
   deployer : SignerWithAddress,
-  accessControllerAddress : string
+  accessController : string
```

Comment:
let's keep "address" in the name, so we know this is not a contract object. we will call `accessController` the actual contract object. this way we know easily what is needed